### ISSUE TITLE
Changing RuleEngine thread pool value ignored 2.4.0

### ISSUE THREAD

**Issue by lukicsl** [openhab/openhab-core#571]
**Expected:**

When setting runtime.cfg:
org.eclipse.smarthome.threadpool:RuleEngine=30

Up to 30 parallel thread are executing ongoing rule execution

**Tested:**

Only 5 threads are executed in parallel

**Setup:**

1. Change runtime.cfg and restart OH
```
org.eclipse.smarthome.threadpool:RuleEngine=30
org.eclipse.smarthome.threadpool:ruleEngine=30
```

2. Verification
```
openhab> config:list "(service.pid=org.eclipse.smarthome.threadpool)"
----------------------------------------------------------------
Pid:            org.eclipse.smarthome.threadpool
BundleLocation: null
Properties:
   RuleEngine = 30
   discovery = 5
   homematicGateway = 30
   homematicRpc = 30
   safeCall = 10
   service.pid = org.eclipse.smarthome.threadpool
   thingHandler = 5
openhab>           

```
3. define items and rules
`Switch   ToggleProxy     "Toggle  Proxy"      ` 

```
rule "Toggle Runner"
when 
        Time cron "0/1 1/1 * 1/1 * ? *" 

then
	logInfo("Toggle Runner", "toggle executed at cron 0 0/1 * 1/1 * ? *" )
        sendCommand(ToggleProxy, ON)
        Thread::sleep(10) 
        sendCommand(ToggleProxy, OFF)

end

rule "Toggle Executer 1"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 1" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 2"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 2" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 3"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 3" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 4"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 4" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 5"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 5" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 6"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 6" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 7"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 7" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 8"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 8" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 9"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 9" )
        Thread::sleep(5000) 

end

rule "Toggle Executer 10"
when 
	Item ToggleProxy  changed from OFF to ON 
then
	logInfo("Toggle Executer", "ON for 10" )
        Thread::sleep(5000) 

end

```
**Result**

Always only 5 threads are running

```
openhab> shell:threads --list |grep -i "RuleEngine"                                                                                                                                                                
779  │ ESH-ruleEngine-9                                                                                           │ TIMED_WAITING │ 32361    │ 21180
788  │ ESH-ruleEngine-10                                                                                          │ TIMED_WAITING │ 32253    │ 21070
789  │ ESH-ruleEngine-11                                                                                          │ TIMED_WAITING │ 32168    │ 21090
790  │ ESH-ruleEngine-12                                                                                          │ TIMED_WAITING │ 32466    │ 21260
791  │ ESH-ruleEngine-13                                                                                          │ TIMED_WAITING │ 32206    │ 21090
1085 │ pipe-grep -i "RuleEngine"                                                                                  │ RUNNABLE      │ 16       │ 10
openhab>  

```
log output
```
2019-02-12 16:12:32.108 [INFO ] [arthome.model.script.Toggle Executer] - ON for 4
2019-02-12 16:12:32.278 [INFO ] [arthome.model.script.Toggle Executer] - ON for 6
2019-02-12 16:12:32.295 [INFO ] [arthome.model.script.Toggle Executer] - ON for 10
2019-02-12 16:12:32.376 [INFO ] [arthome.model.script.Toggle Executer] - ON for 1
2019-02-12 16:12:32.382 [INFO ] [arthome.model.script.Toggle Executer] - ON for 2

2019-02-12 16:12:37.130 [INFO ] [arthome.model.script.Toggle Executer] - ON for 3
2019-02-12 16:12:37.323 [INFO ] [arthome.model.script.Toggle Executer] - ON for 7
2019-02-12 16:12:37.334 [INFO ] [arthome.model.script.Toggle Executer] - ON for 9
2019-02-12 16:12:37.410 [INFO ] [arthome.model.script.Toggle Executer] - ON for 5
2019-02-12 16:12:37.414 [INFO ] [arthome.model.script.Toggle Executer] - ON for 8

2019-02-12 16:12:42.148 [INFO ] [arthome.model.script.Toggle Executer] - ON for 4
2019-02-12 16:12:42.348 [INFO ] [arthome.model.script.Toggle Executer] - ON for 6
2019-02-12 16:12:42.376 [INFO ] [arthome.model.script.Toggle Executer] - ON for 10
2019-02-12 16:12:42.430 [INFO ] [arthome.model.script.Toggle Executer] - ON for 1
2019-02-12 16:12:42.439 [INFO ] [arthome.model.script.Toggle Executer] - ON for 2

2019-02-12 16:12:47.173 [INFO ] [arthome.model.script.Toggle Executer] - ON for 3
2019-02-12 16:12:47.383 [INFO ] [arthome.model.script.Toggle Executer] - ON for 7
2019-02-12 16:12:47.398 [INFO ] [arthome.model.script.Toggle Executer] - ON for 9
2019-02-12 16:12:47.453 [INFO ] [arthome.model.script.Toggle Executer] - ON for 5
2019-02-12 16:12:47.470 [INFO ] [arthome.model.script.Toggle Executer] - ON for 8

2019-02-12 16:12:52.222 [INFO ] [arthome.model.script.Toggle Executer] - ON for 4
2019-02-12 16:12:52.425 [INFO ] [arthome.model.script.Toggle Executer] - ON for 6
2019-02-12 16:12:52.448 [INFO ] [arthome.model.script.Toggle Executer] - ON for 10
2019-02-12 16:12:52.473 [INFO ] [arthome.model.script.Toggle Executer] - ON for 1
2019-02-12 16:12:52.491 [INFO ] [arthome.model.script.Toggle Executer] - ON for 2

```
every 5s a set of 5 log entries are generated

**Comment by openhab-5iver** [comment]
Try removing the first line. The option is 'ruleEngine' and having the first line may be confusing things. I haven't tested this recently, but it was [fixed back in October](https://github.com/eclipse/smarthome/issues/6289).

**Comment by lukicsl** [comment]
Removed the first line and restarted.
Same result:
```
org.eclipse.smarthome.threadpool:ruleEngine=30

openhab> config:list "(service.pid=org.eclipse.smarthome.threadpool)"                                                                                                                                                                                                         
----------------------------------------------------------------
Pid:            org.eclipse.smarthome.threadpool
BundleLocation: null
Properties:
   RuleEngine = 30
   discovery = 5
   homematicGateway = 30
   homematicRpc = 30
   safeCall = 10
   service.pid = org.eclipse.smarthome.threadpool
   thingHandler = 5
openhab> shell:threads --list |grep -i "RuleEngine"                                                                                                                                                                                                                           
127 │ ESH-ruleEngine-1                                                                                          │ TIMED_WAITING │ 965      │ 680
148 │ ESH-ruleEngine-2                                                                                          │ TIMED_WAITING │ 1343     │ 890
158 │ ESH-ruleEngine-3                                                                                          │ TIMED_WAITING │ 1194     │ 810
186 │ ESH-ruleEngine-4                                                                                          │ TIMED_WAITING │ 1173     │ 780
187 │ ESH-ruleEngine-5                                                                                          │ TIMED_WAITING │ 1075     │ 720
318 │ pipe-grep -i "RuleEngine"                                                                                 │ TIMED_WAITING │ 30       │ 20
openhab>   

```

**Comment by lukicsl** [comment]
different results on 2.5.0-snapshot:

```
  ____  ____  ___  ____  / / / /   |  / __ )     
 / __ \/ __ \/ _ \/ __ \/ /_/ / /| | / __  | 
/ /_/ / /_/ /  __/ / / / __  / ___ |/ /_/ /      
\____/ .___/\___/_/ /_/_/ /_/_/  |_/_____/     
    /_/                        2.5.0-SNAPSHOT
                               Build #1521   

Hit '<tab>' for a list of available commands
and '[cmd] --help' for help on a specific command.
Hit '<ctrl-d>' or type 'system:shutdown' or 'logout' to shutdown openHAB.

openhab> config:list "(service.pid=org.eclipse.smarthome.threadpool)"                                                                                                                                                                                                         
----------------------------------------------------------------
Pid:            org.eclipse.smarthome.threadpool
BundleLocation: null
Properties:
   discovery = 5
   homematicGateway = 30
   homematicRpc = 30
   ruleEngine = 30
   safeCall = 10
   service.pid = org.eclipse.smarthome.threadpool
   thingHandler = 5
openhab> 
``` 

```
openhab> shell:threads --list |grep -i "RuleEngine"                                                                                                                                                                                                                           
189 │ ESH-ruleEngine-2                                                                              │ TIMED_WAITING │ 1027     │ 730
190 │ ESH-ruleEngine-3                                                                              │ TIMED_WAITING │ 949      │ 680
192 │ ESH-ruleEngine-4                                                                              │ TIMED_WAITING │ 989      │ 710
193 │ ESH-ruleEngine-5                                                                              │ TIMED_WAITING │ 1021     │ 750
195 │ ESH-ruleEngine-6                                                                              │ TIMED_WAITING │ 971      │ 700
218 │ ESH-ruleEngine-7                                                                              │ TIMED_WAITING │ 542      │ 390
219 │ ESH-ruleEngine-8                                                                              │ TIMED_WAITING │ 590      │ 440
220 │ ESH-ruleEngine-9                                                                              │ TIMED_WAITING │ 614      │ 450
221 │ ESH-ruleEngine-10                                                                             │ TIMED_WAITING │ 548      │ 390
222 │ ESH-ruleEngine-11                                                                             │ TIMED_WAITING │ 526      │ 390
223 │ ESH-ruleEngine-12                                                                             │ TIMED_WAITING │ 555      │ 390
224 │ ESH-ruleEngine-13                                                                             │ TIMED_WAITING │ 512      │ 380
225 │ ESH-ruleEngine-14                                                                             │ TIMED_WAITING │ 550      │ 420
226 │ ESH-ruleEngine-15                                                                             │ TIMED_WAITING │ 531      │ 410
227 │ ESH-ruleEngine-16                                                                             │ TIMED_WAITING │ 604      │ 440
228 │ ESH-ruleEngine-17                                                                             │ TIMED_WAITING │ 787      │ 550
229 │ ESH-ruleEngine-18                                                                             │ TIMED_WAITING │ 493      │ 350
230 │ ESH-ruleEngine-19                                                                             │ TIMED_WAITING │ 516      │ 380
231 │ ESH-ruleEngine-20                                                                             │ TIMED_WAITING │ 541      │ 400
232 │ ESH-ruleEngine-21                                                                             │ TIMED_WAITING │ 506      │ 360
233 │ ESH-ruleEngine-22                                                                             │ TIMED_WAITING │ 569      │ 410
234 │ ESH-ruleEngine-23                                                                             │ TIMED_WAITING │ 541      │ 390
235 │ ESH-ruleEngine-24                                                                             │ TIMED_WAITING │ 501      │ 370
236 │ ESH-ruleEngine-25                                                                             │ TIMED_WAITING │ 515      │ 380
237 │ ESH-ruleEngine-26                                                                             │ TIMED_WAITING │ 518      │ 390
238 │ ESH-ruleEngine-27                                                                             │ TIMED_WAITING │ 494      │ 370
239 │ ESH-ruleEngine-28                                                                             │ TIMED_WAITING │ 502      │ 360
240 │ ESH-ruleEngine-29                                                                             │ TIMED_WAITING │ 529      │ 380
241 │ ESH-ruleEngine-30                                                                             │ TIMED_WAITING │ 500      │ 380
242 │ ESH-ruleEngine-31                                                                             │ TIMED_WAITING │ 526      │ 390
271 │ pipe-grep -i "RuleEngine"                                                                     │ RUNNABLE      │ 28       │ 20
openhab> 
```
Fix seems did not found its way to 2.4.0 final
